.github/workflows/full_stack_tests.yml

```diff
@@ -0,0 +1,512 @@
+        uses: ./.github/actions/install-oppia-dependencies
+      - name: Build Webpack
+        run: python -m scripts.build --prod_env
+      - name: Zip build files
+        # We avoid using ../ or absolute paths because unzip treats these as
+        # security issues and will refuse to follow them.
+        run: |
+          zip -rqy build_files.zip oppia/third_party oppia_tools oppia/build oppia/webpack_bundles oppia/app.yaml oppia/assets/hashes.json oppia/backend_prod_files oppia/dist
+        working-directory: /home/runner/work/oppia
+      - name: Upload build files artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: build_files
+          path: /home/runner/work/oppia/build_files.zip
+          retention-days: 1
+  e2e_test:
+    needs: [check_test_suites_to_run, build]
+    runs-on: ubuntu-22.04
+    if: ${{ fromJSON(needs.check_test_suites_to_run.outputs.TEST_SUITES_TO_RUN).e2e.count > 0 }}
+    strategy:
+      max-parallel: 3
+      matrix:
+        suite: ${{ fromJSON(needs.check_test_suites_to_run.outputs.TEST_SUITES_TO_RUN).e2e.suites }}
+    name: E2E (${{ matrix.suite.name }})
+    steps:
+      - name: Checkout repository so that local actions can be used
+        uses: actions/checkout@v4
+      - name: Merge develop and set up Python
+        uses: ./.github/actions/merge-develop-and-setup-python
+        with:
+          use_cache: true
+      - name: Cache node modules
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: /home/runner/work/oppia/yarn_cache
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - name: Download build files artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: build_files
+          path: /home/runner/work/oppia
+      - name: Unzip build files
+        run: |
+          pwd
+          ls -la
+          unzip build_files.zip
+          rm build_files.zip
+          echo "Files in ./:"
+          ls -la .
+          echo "Files in oppia_tools:"
+          ls -la oppia_tools
+          echo "Files in oppia:"
+          ls -la oppia
+          echo "Files in build:"
+          ls -la oppia/build
+          echo "Files in third_party:"
+          ls -la oppia/third_party
+        working-directory: /home/runner/work/oppia
+        shell: bash
+      - name: Install Oppia dependencies
+        uses: ./.github/actions/install-oppia-dependencies
+      - name: Install Chrome
+        uses: ./.github/actions/install-chrome
+      - name: Install ffmpeg for wdio videos
+        run: |
+          sudo apt-get update
+          sudo apt install ffmpeg
+      - name: Run E2E Test ${{ matrix.suite.name }}
+        uses: oppia/retry@a9fb265410e634c84ee90f3e87f323fde0541037
```

Comment:
since we already have a fork of the `retry` action, can we use our fork elsewhere where we use `nick-fields/retry`?

Reply:
I'm not sure what's best here, actually. The fork has some "flaky execution substrings" modifications that are only applicable to e2e tests, so I feel like this is really the only place where it's applicable. Maybe we can discuss this briefly when we chat tomorrow...

Reply:
decided to leave this as-is until we decide how to keep forks like `oppia/retry` updated and whether they're worth maintaining